ACCESS-ESM1.5 QA test: CICE icefields_nml configuration

jo-basevi opened the pull request:
Added QA test for separate `icefields_nml` in `ice_history.nml`  (in as described in #50)

Closes #50
Closes #44

jo-basevi commented:
I quickly added a generic test for the ACCESS-ESM1.5 specific config tests - to just checking the tests pass without error (and also to pass the codecov CI checks..)

I ran some manual tests with an configuration to test the error messages, and that they failed when expected

CodeGat reviewed (dismissed):
Looks good to me! 

Thanks for fixing up the codecov tests...the red crosses were a bit scary when I was doing the initial esm1.5 QA checks

anton-seaice commented on src/model_config_tests/qa/test_access_esm1p5_config.py:
```diff
@@ -191,3 +197,34 @@ def test_mppnccombine_fast_collate_exe(self, config):
+        cice_in_path = cice_control_path / CICE_IN_NML_FNAME
+        assert cice_in_path.is_file()
+        cice_in = f90nml.read(cice_in_path)
+        assert ICEFIELDS_NML_NAME not in cice_in, icefields_nml_error_msg
```
We could also check there are no repeated fields between the two nml files

jo-basevi replied:
So for example checking fields in `&setup_nml` are different between `cice_in.nml` and `ice_history.nml`? If so, added in e92cad4.

anton-seaice replied:
Looks good - thanks!

anton-seaice reviewed (commented):
It seems like there are three things bundled into one PR here ? 

1. Adding a qa test for cice namelist
2. rearranging the directories
3. add a config for preindustrial+concentrations

In future we should do this as seperate PRs so we can track the changes easier and review it easier !

jo-basevi commented:
> It seems like there are three things bundled into one PR here ?

Yes, sorry about that! I added the tests so they covered the lines I added in the first commit. But yeah next time, I'll add the wider tests for tests in an earlier PR.

anton-seaice commented on src/model_config_tests/qa/test_access_esm1p5_config.py:
```diff
@@ -191,3 +197,44 @@ def test_mppnccombine_fast_collate_exe(self, config):
+        cice_in_path = cice_control_path / CICE_IN_NML_FNAME
+        assert cice_in_path.is_file()
+        cice_in = f90nml.read(cice_in_path)
+        assert ICEFIELDS_NML_NAME not in cice_in, icefields_nml_error_msg
```
```suggestion
        assert ICEFIELDS_NML_NAME not in cice_in, icefields_nml_wrong_file_error_msg
```
Sorry to be a pedant, this should have a different error message. e.g.:

"icefields_nml namelist found in cice_in.nml. This should only be in ice_history.nml to prevent duplication."

jo-basevi replied:
Thanks, that does make it more clear! I've added that in 0e91aec.

anton-seaice commented on src/model_config_tests/qa/test_access_esm1p5_config.py:
```diff
@@ -191,3 +197,44 @@ def test_mppnccombine_fast_collate_exe(self, config):
+
+        # Check icefields_nml not in cice_in.nml
+        cice_in_path = cice_control_path / CICE_IN_NML_FNAME
+        assert cice_in_path.is_file()
```
Does this need an error message? I guess its fairly self explanatory

jo-basevi replied:
Yeah a more descriptive error message wouldn't hurt.. I've added that in 0e91aec. It is listed as a required config file in the payu model driver

anton-seaice reviewed (dismissed):
Thanks Jo! - I ran some adhoc tests using `dev-historical+concentrations` from `https://github.com/ACCESS-NRI/access-esm1.5-configs`

- The default passes, 
- adding &icefields_nml to cice_in gives a positive failure, but the error message could be better,
- adding fields to &setup_nml detects gives a positive failure for duplicated fields (correct)
- renaming ice_history.nml & cice_in.nml both fail correctly

anton-seaice reviewed (approved):
Looks good Jo

CodeGat commented:
This should also close #44

jo-basevi commented:
Thanks all for the reviews!
